Source/WebCore/html/CanvasBase.cpp

```diff
@@ -253,18 +253,19 @@ RefPtr<ImageBuffer> CanvasBase::setImageBuffer(RefPtr<ImageBuffer>&& buffer) con
     IntSize oldSize = m_size;
     size_t oldMemoryCost = m_imageBufferMemoryCost.load(std::memory_order_relaxed);
     size_t newMemoryCost = 0;
-    if (m_imageBuffer) {
-        m_size = m_imageBuffer->truncatedLogicalSize();
-        newMemoryCost = m_imageBuffer->memoryCost();
-        m_imageBuffer->context().setShadowsIgnoreTransforms(true);
-        m_imageBuffer->context().setImageInterpolationQuality(defaultInterpolationQuality);
-        m_imageBuffer->context().setStrokeThickness(1);
-        m_contextStateSaver = makeUnique<GraphicsContextStateSaver>(m_imageBuffer->context());
+    if (RefPtr imageBuffer = m_imageBuffer) {
+        m_size = imageBuffer->truncatedLogicalSize();
+        newMemoryCost = imageBuffer->memoryCost();
+        imageBuffer->context().setShadowsIgnoreTransforms(true);
+        imageBuffer->context().setImageInterpolationQuality(defaultInterpolationQuality);
+        imageBuffer->context().setStrokeThickness(1);
+        m_contextStateSaver = makeUnique<GraphicsContextStateSaver>(imageBuffer->context());
     }
     m_imageBufferMemoryCost.store(newMemoryCost, std::memory_order_relaxed);
     if (newMemoryCost) {
-        JSC::JSLockHolder lock(scriptExecutionContext()->vm());
-        scriptExecutionContext()->vm().heap.reportExtraMemoryAllocated(static_cast<JSCell*>(nullptr), newMemoryCost);
+        RefPtr scriptExecutionContext = this->scriptExecutionContext();
+        JSC::JSLockHolder lock(scriptExecutionContext->vm());
```

Comment:
This is the only usage within the method, so you can do this:
```suggestion
        JSC::JSLockHolder lock(protectedScriptExecutionContext()->vm());
```